jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/MultiPartByteRanges.java

```diff
@@ -132,6 +135,34 @@ public boolean addPart(MultiPart.Part part)
         }
     }
 
+    /**
+     * <p>A specialized {@link org.eclipse.jetty.io.content.InputStreamContentSource}
+     * whose content is sliced by a byte range.</p>
+     */
+    public static class InputStreamContentSource extends org.eclipse.jetty.io.content.InputStreamContentSource
+    {
+        private long toRead;
+
+        public InputStreamContentSource(InputStream inputStream, ByteRange byteRange) throws IOException
+        {
+            super(inputStream);
```

Comment:
Can you put a `requiresNotNull` in the constructor of `InputStreamContentSource` just in case a `Resource` has a null input stream.

Reply:
Done, but in the parent class.